lib/authorizer/oauth2.js

```diff
@@ -58,7 +59,46 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done -
      */
     pre: function (auth, done) {
-        done(null, Boolean(auth.get('accessToken')));
+        const authId = auth.get('authId'),
+            refreshTokenHelper = auth.get('refreshTokenHelper');
+
+        // Go into the refresh token flow ONLY IF we have all necessary information present
+        if (!(authId && refreshTokenHelper && _.isFunction(refreshTokenHelper.refreshToken))) {
+            return done(null, Boolean(auth.get('accessToken')));
+        }
+
+        let refreshTokenObject = refreshTokenHelper.refreshToken(authId),
+            refreshTokenTimeout = refreshTokenHelper.timeout,
+            refreshPromises = [];
+
+        if (!refreshTokenObject || !refreshTokenObject.promise) {
```

Comment:
Let's consistently return a promise to avoid this block. In case of error we can return a rejected promise.